tests/unit/auth/v2/queryAuthCheck.js

```diff
@@ -26,3 +27,94 @@ describe('v2: queryAuthCheck', () => {
         }
     }));
 });
+
+describe('v2: queryAuthCheck', () => {
+    let clock;
+
+    beforeEach(() => {
+        clock = sinon.useFakeTimers();
+    });
+    afterEach(() => {
+        process.env.PRE_SIGN_URL_EXPIRY = 604800000;
+        clock.restore();
+    });
+    it('URL should not expire before 7 days with default expiry', () => {
+        const currentTime = Date.now() / 1000;
+        const expires = currentTime + 604799;  // in seconds
+        const mockRequest = {
+            method: 'GET',
+            url: 'mockurl',
+            query: {
+                Expires: expires,
+            },
+            headers: {
+                'Content-MD5': 'c',
+            },
+        };
+        const data = {
+            Expires: expires,
+            AWSAccessKeyId: 'keyId',
+            Signature: 'sign',
+        };
+        const res = queryAuthCheck(mockRequest, log, data);
+        assert.notStrictEqual(res.err.AccessDenied, true);
+        assert.notStrictEqual(res.err.RequestTimeTooSkewed, true);
+    });
+    it('URL should expire after 7 days with default expiry', () => {
+        clock.tick(604800000);  // take time 604800000ms (7 days) ahead
+        const currentTime = Date.now();
+        const request = { method: 'GET', query: { Expires: currentTime } };
+        const data = { Expires: currentTime };
+        const res = queryAuthCheck(request, log, data);
+        assert.notStrictEqual(res.err, undefined);
```

Comment:
minor style: might be slightly more readable to do `assert(res.err)` and it does a truish test, as some callbacks return `null` as the error object (not `undefined`), it can be a more generic way to check that an error occurred.

Reply:
TBH, it bothered me as well :) I thought it could be good to have a helper util such as `nullOrUndefined` or something like that - maybe another to check for even more falsey values depending on the use case... For now, I added additional check for the `null` case to cover more potential scenarios.

Reply:
May be I am wrong, I don't see `queryAuthCheck` returning `err: undefined`!